File: backends/java/zappy-application/src/main/java/com/zappymart/application/ports/OrderRepository.java
package com.zappymart.application.ports;

import com.zappymart.domain.ordering.Order;

import java.util.List;
import java.util.Optional;

public interface OrderRepository {

    Order save(Order order);

    List<Order> ofCustomerNewestFirst(String customerId);

    Optional<Order> byIdOfCustomer(String orderId, String customerId);

    String nextOrderNumber();
}

File: backends/kotlin/zappy-application/src/main/kotlin/nl/zappymart/application/ports/OrderRepository.kt
package nl.zappymart.application.ports

import nl.zappymart.domain.ordering.Order

interface OrderRepository {

    fun save(order: Order): Order

    fun page(customerId: String, size: Int, afterOrderId: String?): List<Order>

    fun count(customerId: String): Int

    fun findForCustomer(orderId: String, customerId: String): Order?
}
